# 修正ログ

* SSLでなくSSH
* UserRegister.py
    * 起動エラー（Database.pyコンストラクタ引数変更の影響）
    * `insert`サブコマンドで `-s` フラグなし時にフリーズする（SSH通信応答返らず）
    * `sqlite3`コマンドがない（pythonのsqlite3だけで実行できるようコード変更したい）

## SSLでなくSSH

恥ずかしいミス。変更対象ファイルは以下。

* ./res/config.ini
* ./src/setting/Setting.py
* ./src/cui/uploader/command/repository/Creator.py
* ./src/cui/register/command/Inserter.py

## UserRegister.pyが起動エラー

前回修正時、Database.pyの引数をなくしたのが原因。呼出でエラーになる。

python /home/pi/root/script/py/GitHub.Uploader.Pi3.Https.201802210700/src/UserRegister.py insert -u ytmemo -p takasa02080

python /tmp/GitHub.Uploader.Pi3.Https.201802212000/src/UserRegister.py insert -u ytmemo -p takasa02080

find . -name "*.py" -print0 | xargs -0 grep "database.Database.Database(self."

database.Database.Database(self.__path_dir_root)

find . -name "*.py" -print0 | xargs -0 sed -i.bak "s/database.Database.Database(self.__path_dir_root)/database.Database.Database()/g"

find . -type f -name "*.bak" -print0 | xargs -0 rm -f

## insert -s フラグなし時にフリーズする

SSH用に秘密鍵と公開鍵を作成した後、`ssh -T git@github.com.{user}`するが、ここでフリーズする。

ラズパイではなぜか上記コマンドのときにフリーズする。応答が返ってこない。

以下のように対応する。`./res/config.ini`の`[Git]Remote`値が`SSL`のときだけ上記コマンドを実行する。
```python
self.__setting = setting.Setting.Setting()
if 'SSL' == self.__setting.GitRemote:
    # 3A-3. SSH接続確認
    self.__sshkeygen.CheckSshConnect(host, args.username)
else:
    # ラズパイでなぜかSSH接続確認するとフリーズするので省略。
    print('SSH接続確認を省略します。')
```

## `sqlite3`コマンドがない

ターミナルコマンドで実行している。だがラズパイには無い。初期DBを作るためにcreate文やinsert文の発行をしているだけ。インストールせねばならないのが面倒。

pythonに標準装備のsqlite3を使えばいけるはず。

またはdatasetのDatabase.query()を使えば。
```sh
result = db.query(statement)
```

`./src/database/`配下の各所をすべて変更せねばならない！大量！

UserRegister.pyを実行するには複数DBを修正せねばならない。結合テストもそれからでないとできない。

* 修正完了
    * `./src/database/account/Main.py`

